compiler/rustc_mir_transform/src/validate.rs

```diff
@@ -586,6 +589,22 @@ impl<'a, 'tcx> TypeChecker<'a, 'tcx> {
 
         crate::util::relate_types(self.tcx, self.param_env, variance, src, dest)
     }
+
+    /// Check that the given predicate definitely holds in the param-env of this MIR body.
```

Comment:
Do we already *have* a "test this predicate holds" function somewhere in the validator? I couldn't find one when doing a simple search.